File: tests/machine/x/dart/join_multi.dart
void main() {
  var customers = [
    {'id': 1, 'name': 'Alice'},
    {'id': 2, 'name': 'Bob'},
  ];
  var orders = [
    {'id': 100, 'customerId': 1},
    {'id': 101, 'customerId': 2},
  ];
  var items = [
    {'orderId': 100, 'sku': 'a'},
    {'orderId': 101, 'sku': 'b'},
  ];
  var result = (() {
    var _q0 = <dynamic>[];
    for (var o in orders) {
      _q0.add({'name': c.name, 'sku': i.sku});
    }
    return _q0;
  })();
  print('--- Multi Join ---');
  var _iter1 = result;
  for (var r in (_iter1 is Map ? (_iter1 as Map).keys : _iter1) as Iterable) {
    print([r['name'], 'bought item', r['sku']].join(' '));
  }
}

File: tests/machine/x/go/join_multi.go
//go:build ignore

package main

import (
	"fmt"
)

func main() {
	var customers []CustomersItem = []CustomersItem{CustomersItem{
		1,
		"Alice",
	}, CustomersItem{
		2,
		"Bob",
	}}
	_ = customers
	var orders []OrdersItem = []OrdersItem{OrdersItem{
		100,
		1,
	}, OrdersItem{
		101,
		2,
	}}
	var items []ItemsItem = []ItemsItem{ItemsItem{
		100,
		"a",
	}, ItemsItem{
		101,
		"b",
	}}
	_ = items
	var result []Result = func() []Result {
		results := []Result{}
		for _, o := range orders {
			for _, c := range customers {
				if !(o.CustomerId == c.Id) {
					continue
				}
				for _, i := range items {
					if !(o.Id == i.OrderId) {
						continue
					}
					results = append(results, Result{
						c.Name,
						i.Sku,
					})
				}
			}
		}
		return results
	}()
	fmt.Println("--- Multi Join ---")
	for _, r := range result {
		fmt.Println(r.Name, "bought item", r.Sku)
	}
}
